subsys/net/lib/aws_jobs/src/aws_jobs.c

```diff
@@ -0,0 +1,360 @@
+/*
+ *copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/mqtt.h>
+#include <logging/log.h>
+#include "aws_jobs.h"
+
+LOG_MODULE_REGISTER(aws_jobs, CONFIG_AWS_JOBS_LOG_LEVEL);
+
+
+/**
+ * @brief Local function used to error check snprintf outputs.
+ *
+ * @param[in] ret Return value of snpritnf.
+ * @param[in] size Max size of what would have been the final string.
+ * @param[in] entry Name of the function it was called from.
+ *
+ * @retval 0 If sucessful otherwise -ENOMEM if the buffer was too small for the
+ *	string or the error code of snprintf if the formating failed.
+ */
+static inline int report_snprintf_err(u32_t ret, u32_t size, char *entry)
+{
+	if (ret >= size) {
+		LOG_ERR("Unable to fit formated string into to allocate"
+			"memory for %s", entry);
+		return -ENOMEM
+	} else if (ret < 0) {
+		LOG_ERR("Output error for %s was encountered with return value"
+			" %d", entry, ret);
+		return ret;
+	}
+	return 0;
+}
+
+
+/**
+ * @brief Constructs the get topic for accepted and rejeceted AWS jobs.
+ *
+ * @param[in] client_id Client ID of the current connected MQTT session.
+ * @param[out] topic Outputs the constructed mqtt_topic into the given buffer.
+ *
+ * @retval 0 If sucessful otherwise the return value of report_snprintf_err
+ */
+static int construct_get_topic(u8_t *client_id, struct mqtt_topic *topic)
+{
+	char get_topic[GET_TOPIC_LEN + 1];
+	int ret = snprintf(get_topic,
+			   GET_TOPIC_LEN,
+			   GET_TOPIC_TEMPLATE,
+			   client_id,
+			   "#");
+	ret = report_snprintf_err(ret, GET_TOPIC_LEN, "get_topic");
+
+	if (ret) {
+		return ret;
+	}
+
+	topic->topic.utf8 = get_topic;
+	topic->topic.size = strlen(get_topic);
+	topic->qos = MQTT_QOS_1_AT_LEAST_ONCE;
+	return 0;
```

Comment:
And at this point, `get_topic` is gone. We can't use a stack variable for the buffer, at least not within this helper function.

I see that this issue happens in other places as well, it should all be fixed.

Reply:
Great spot didn't think about that, will fix it.

Reply:
Think this should be fixed now. If not reopen the conversation.

Reply:
Doesn't look fixed.